client/deposits/details/index.tsx

```diff
@@ -142,16 +129,85 @@ export const DepositOverview: React.FC< DepositOverviewProps > = ( {
 				)
 			}
 			value={ <DepositStatusIndicator deposit={ deposit } /> }
-			detail={ deposit.bankAccount }
+			detail={
+				<>
+					{ deposit.bankAccount }
+					<br />
+					Bank reference ID:{ ' ' }
+					{ deposit.bank_reference_key ? (
+						<>
+							<span className="woopayments-payout-bank-reference-id">
+								{ deposit.bank_reference_key }
+							</span>
+							<button
+								type="button"
+								className={ `woopayments-payout-bank-reference-id-copy-button ${
+									isCopied ? 'state--copied' : ''
+								}` }
+								aria-label={ __(
+									'Copy bank reference ID to clipboard',
+									'woocommerce-payments'
+								) }
+								title={ __(
+									'Copy to clipboard',
+									'woocommerce-payments'
+								) }
+								onClick={ () => {
+									const bankReferenceId = document.querySelector(
+										'.woopayments-payout-bank-reference-id'
+									)?.textContent;
+
+									if ( bankReferenceId ) {
+										navigator.clipboard.writeText(
+											bankReferenceId
+										);
+									}
+
+									if ( ! isCopied ) {
+										setIsCopied( true );
+										setTimeout( () => {
```

Comment:
Need to take care with `setTimeout` in React. In particular, need to clean up the timeout on unmount. (I need a refresher on this myself).

This article looks reasonable, there may be better info out there (especially on React official docs): https://felixgerschau.com/react-hooks-settimeout/

Reply:
Very helpful and a great concept I learned today! I have implemented it in the CopyButton component. I hope I have done it correctly. Would appreciate if you could check once 🙏🏼